src/test/java/org/javarosa/core/storage/IndexedStorageUtilityTests.java

```diff
@@ -0,0 +1,155 @@
+package org.javarosa.core.storage;
+
+import org.javarosa.core.services.storage.IStorageUtilityIndexed;
+import org.javarosa.core.services.storage.util.DummyIndexedStorageUtility;
+import org.javarosa.core.util.Interner;
+import org.javarosa.core.util.externalizable.LivePrototypeFactory;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Vector;
+
+/**
+ * Tests for basic storage utility functions
+ *
+ * Created by ctsims on 9/22/2017.
+ */
+
+public class IndexedStorageUtilityTests {
```

Comment:
I actually wrote this to directly test the mockstorage itself, although I agree that it would be clearer to have the classes split out, I'll go ahead and do that.